### PR TITLE
mgr/cephadm: Fix haproxy not being recognized as a proper daemon

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Turns out daemon types != service types: 

    cephadm [WRN] Found unknown service type haproxy on host smithi019
    cephadm [WRN] Found unknown service type keepalived on host smithi019

leading to `self.mgr.cache.get_daemons_by_service(spec.service_name())` 
not returning any daemons. 

Fixes: https://tracker.ceph.com/issues/51311

Signed-off-by: Sebastian Wagner <sewagner@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/services/ingress.py [hunk 1]
```diff
@@ -183,6 +184,11 @@ def keepalived_generate_config(
             password = spec.keepalived_password
 
         daemons = self.mgr.cache.get_daemons_by_service(spec.service_name())
+
+        if not daemons:
+            raise OrchestratorError(
+                f'Failed to generate keepalived.conf: Do daemons deployed for {spec.service_name()}')
```
Comment (asannes) [inline src/pybind/mgr/cephadm/services/ingress.py:190]: Typo? -Do +No ?

**Review by adk3798 — APPROVED** [review]
LGTM minus the typo already pointed out

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-07-21_15:02:39-rados:cephadm-wip-swagner3-testing-2021-07-21-1416-distro-basic-smithi/6284047/

**Comment by tchaikov** [comment]
jenkins test make check

**Pull request merged**